core/src/main/java/de/unistuttgart/iaas/amyassist/amy/core/natlang/NLProcessingManager.java

```diff
@@ -45,4 +45,10 @@
 	 */
 	void decideIntent(DialogImpl dialog, String naturalLanguageText);
 
+	/**
+	 * @param dialog
+	 * @param naturalLanguageText
+	 */
+	void processIntent(DialogImpl dialog, String naturalLanguageText);
```

Comment:
use the `Dialog` interface instead of the imlementation

Reply:
this would require a cast - i might find another solution

Reply:
define all required methods on the interface